interfaces/apparmor/backend.go

```diff
@@ -414,16 +414,36 @@ func addUpdateNSProfile(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 }
 
 func addContent(securityTag string, snapInfo *snap.Info, opts interfaces.ConfinementOptions, snippetForTag string, content map[string]*osutil.FileState) {
-	var policy string
+	// Normally we use a specific apparmor template for all snap programs.
+	policy := defaultTemplate
+	ignoreSnippets := false
+	// Classic confinement (unless overridden by JailMode) has a dedicated
+	// permissive template that applies a strict, but very open, policy.
+	if opts.Classic && !opts.JailMode {
+		policy = classicTemplate
+		ignoreSnippets = true
+	}
 	// When partial AppArmor is detected, use the classic template for now. We could
 	// use devmode, but that could generate confusing log entries for users running
 	// snaps on systems with partial AppArmor support.
-	level := release.AppArmorLevel()
-	if level == release.PartialAppArmor || (opts.Classic && !opts.JailMode) {
-		policy = classicTemplate
-	} else {
-		policy = defaultTemplate
+	if level := release.AppArmorLevel(); level == release.PartialAppArmor {
+		// As a special exception, for openSUSE Tumbleweed which ships Linux
+		// 4.16, do not downgrade confinement template.
+		//
+		// We don't want to do this in general yet because older versions of
+		// the kernel did now provide backwards compatible interpretation of a
+		// confinement so the meaning of the template would change across
+		// kernel versions and we have not validated that the current template
+		// is operational on older kernels.
+		if !release.DistroLike("opensuse-tumbleweed") {
+			policy = classicTemplate
+			ignoreSnippets = true
+		}
 	}
+	// If a snap is in devmode (or is using classic confinement) then make the
+	// profile non-enforcing where violations are logged but not denied.
+	//
+	// XXX: jdstrand, is this desired for classic?)
```

Comment:
Yes this is desired for classic. While the profile is wide open and there *shouldn't* be any policy violations, if there are bugs/etc (ie, revalidation) then we let the profile be in devmode to try not to expose these things to users.